### PR TITLE
Global Styles: Re-add styles that were removed, for classic themes

### LINKED ISSUE #64917: Buttons block: altered styling(colors, padding, border-radius)
### Quick summary

Had a user report that their buttons were missing from multiple sites that are all using the `Canape` theme which seems to be a new bug.

### Steps to reproduce

1. Switch your theme to `Canape` or `Coherent`
2. Make sure that the background color of the theme is set to `white`
3. Add a button block and make sure that the `Style` is set to `Fill` without selecting any colors

### What you expected to happen

The button should turn to a lighter shade of the color specified by the theme.

### What actually happened

The button turns white regardless unless you change the color settings on the individual button (this is also reflected within the Editor as well as you'll see in the video below). You also can't see the button text that you're writing unless the button is not selected.

### Context

35886225-hc

https://user-images.githubusercontent.com/51870544/175398188-17ce808b-6261-42d4-b61e-614c482326eb.mp4

https://user-images.githubusercontent.com/51870544/175398317-2c8bbd4a-44ba-42f7-8da4-af125662b4f2.mp4

### Simple, Atomic or both?

Simple, Atomic

### Theme-specific issue?

So far I've found this happens for both `Canape` and `Coherent`, though it may be affecting others. Does not seem to be affecting and FSE sites based on my testing but I can't guarantee that.

### Browser, operating system and other notes

Consistent across Chrome, Safari, and Firefox

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, easy to implement

### Workaround details

Currently the only consistent workaround I've found that is applicable to both Simple and AT sites is that you need to change the Styles option from `Fill` to `Outline`. There are CSS options, but it seems like this only works if you select the individual element (I've tried several variations of CSS selectors and couldn't get anything to work unless it was specifically called out for a single element).

Video of CSS issue:

https://user-images.githubusercontent.com/51870544/175397009-b8a3f2ff-1831-42ea-8874-c10b072e9f10.mp4

### PR DESCRIPTION
**Pull request by scruffian**
## What?
This is another attempt at https://github.com/WordPress/gutenberg/pull/43981.

In classic themes we don't output the styles for elements, which we rely on for button styles. This adds a classic.scss file that will be output in all classic themes to ensure that those styles are preserved.

Fixes https://github.com/Automattic/wp-calypso/issues/64917

We might decide in the future to create one file per block, but I don't anticipate this being a common problem, so for now I think this solution is sufficient.

## Why?
These styles were removed in https://github.com/WordPress/gutenberg/pull/34180. We explored adding them to all themes in https://github.com/WordPress/gutenberg/pull/43981 but this seems impossible because every theme handles buttons differently.

## Testing Instructions
- Check the visual appearance of buttons blocks in both the frontend and the editor, in both classic and block themes.
- It's worth testing various different themes as they all handle buttons differently!

## Screenshots or screencast 
Screenshot 2022-09-21 at 17 20 09
Screenshot 2022-09-21 at 17 19 50
Screenshot 2022-09-21 at 17 19 16
Screenshot 2022-09-21 at 17 18 47

### CODE REVIEW

**Comment by oandregal** [comment]
I like that this is targeted to only add the styles that are missing.

One thing that gives me pause is this: how should we address the missing styles for any other block that uses the `__experimentalStyles` API (at the moment, navigation & pullquote)? Do we migrate to this file _any_ CSS that uses that API?

**Comment by scruffian** [comment]
I think it would be better to look at that on a case by case basis.

**Comment by scruffian** [comment]
> One thing that gives me pause is this: how should we address the missing styles for any other block that uses the __experimentalStyles API (at the moment, navigation & pullquote)? Do we migrate to this file any CSS that uses that API

These are different because they are block styles, not element styles from the core theme.json.

**Review by oandregal — APPROVED** [review]
This approach works for me, though I'd like to have another pair of eyes to look at this (style engine people, etc.).

The reasons this works for me:

- I've [learned](https://github.com/WordPress/gutenberg/pull/43981#issuecomment-1254826536) that we are already shipping the styles of blocks that use the `__experimentalStyle` API in `block.json` so we only need to address the issue of what to do with _element_ styles that lived in the block-library before ([see](https://github.com/WordPress/gutenberg/pull/43981#issuecomment-1253846432)).
- It does not seem that we'll need to add any more CSS here, as the remaining elements (captions, form elements, etc) do not suffer from the same conditions the block button has (it is also an element).
- Unlike [43981](https://github.com/WordPress/gutenberg/pull/43981), the styles we add here for classic themes are explicit (I wonder if there're any more styles we can migrate here and remove elsewhere).

File: lib/compat/wordpress-6.1/script-loader.php
Comment (ramonjd) [inline lib/compat/wordpress-6.1/script-loader.php]: This hook throws the following notice for me:

```
Notice: Function wp_register_style was called incorrectly. Scripts and styles should not be registered or enqueued until the wp_enqueue_scripts, admin_enqueue_scripts, or login_enqueue_scripts hooks. This notice was triggered by the classic-theme-styles handle. 
```
I've tested the following –

`add_action( 'wp_enqueue_scripts', 'gutenberg_enqueue_classic_theme_styles' );`

`add_action( 'init', 'gutenberg_enqueue_classic_theme_styles' );`

– and both work. Not sure which is better. I suppose I'd lean towards `wp_enqueue_scripts` since it's how we enqueue the other styles?

**Comment by ramonjd** [comment]
> though I'd like to have another pair of eyes to look at this (style engine people, etc.).

From the style engine perspective, I don't think there's much to add. In relation to elements, the style engine is only outputting links in [elements.php](https://github.com/WordPress/gutenberg/blob/7f0a211da6603b336b75b75d8235a50bf547e6f3/lib/block-supports/elements.php).

I've tested in numerous classic and block themes, and the buttons are appearing as I expected. 

I just had a question over the use of the hook `after_setup_theme`, which triggers a notice

```
Notice: Function wp_register_style was called incorrectly. Scripts and styles should not be registered or enqueued until the wp_enqueue_scripts, admin_enqueue_scripts, or login_enqueue_scripts hooks. This notice was triggered by the classic-theme-styles handle. Please see [Debugging in WordPress](https://wordpress.org/support/article/debugging-in-wordpress/) for more information. (This message was added in version 3.3.0.) in /var/www/html/wp-includes/functions.php on line 5839 Notice: Function wp_enqueue_style was called incorrectly. Scripts and styles should not be registered or enqueued until the wp_enqueue_scripts, admin_enqueue_scripts, or login_enqueue_scripts hooks. This notice was triggered by the classic-theme-styles handle. Please see [Debugging in WordPress](https://wordpress.org/support/article/debugging-in-wordpress/) for more information. (This message was added in version 3.3.0.) in /var/www/html/wp-includes/functions.php on line 5839
```

File: lib/compat/wordpress-6.1/script-loader.php
Comment (aristath) [reply lib/compat/wordpress-6.1/script-loader.php]: > Not sure which is better. I suppose I'd lean towards wp_enqueue_scripts since it's how we enqueue the other styles?

Yes, `wp_enqueue_scripts` would be the right hook to use in this scenario 👍

**Review by aristath — APPROVED** [review]
This looks good!
I'll go ahead and pre-approve the PR, pending the change from `after_setup_theme` to `wp_enqueue_scripts` as mentioned in an earlier comment 👍

**Comment by scruffian** [comment]
The problem with `wp_enqueue_scripts` is that it doesn't run in the editor, but I also added an action on `admin_enqueue_scripts` so hopefully that looks ok now?

File: lib/compat/wordpress-6.1/script-loader.php
Comment (getdave) [inline lib/compat/wordpress-6.1/script-loader.php]: As this is unusual, I would favour adding a comment here to explain _why_ we needed the additional `admin_enqueue_scripts`.

Comment (scruffian) [reply lib/compat/wordpress-6.1/script-loader.php]: Done!

**Comment by aristath** [comment]
In that case, maybe `init` would be a better fit? It runs both on the front & dashboard, and we're doing it for other scripts/styles as well 👍

File: lib/compat/wordpress-6.1/script-loader.php
Comment (getdave) [reply lib/compat/wordpress-6.1/script-loader.php]: Sorry my comment wasn't enormously helpful. I could have made a suggestion in Github and done some work for you.

Thanks for following up 👍

**Comment by scruffian** [comment]
> In that case, maybe init would be a better fit? It runs both on the front & dashboard, and we're doing it for other scripts/styles as well 👍

I don't feel strongly, but hooking to two actions seems ok to me.

**Review by andrewserong — APPROVED** [review]
Nice work @scruffian, I think this is the best compromise of all the options, and as both you and @oandregal have mentioned, it ensures that we don't continue to add styles to `theme.json` and accidentally cause issues for Classic themes. The other thing I really like about this approach is that for Classic themes that wish to disable this stylesheet, they can pretty simply dequeue it.

Tested with a range of Classic themes (TwentyTwentyOne, TwentyTwenty, TwentyNineteen,  Canape, Dara, Sequential) and a couple of blocks-based themes to ensure it's all working correctly, and this LGTM! ✨ 

From the perspective of major core releases, I imagine this is a bug fix, so can make it into 6.1? I've added the Bug and Backport labels.

**Comment by scruffian** [comment]
I think this needs to go into 6.1. I started a backport here: https://github.com/WordPress/wordpress-develop/pull/3331

**Comment by ockham** [comment]
I just cherry-picked this PR to the wp/6.1 branch to get it included in the next release: 75786cdfab

**Comment by ethanclevenger91** [comment]
I'm not sure if this is a note for this or for #43981 but I have a "classic theme" with a `theme.json` file where I'm defining button styles via the `elements` section:

```
"styles": {
	"elements": {
		"button": {
			"color": {
				"background": "var(--wp--preset--color--fl-yellow)",
				"text": "white"
			},
			"border": {
				"color": "white",
				"radius": "70px",
				"style": "solid",
				"width": "4px"
			},
			"typography": {
				"fontWeight": "700"
			}
		}
	}
}
```

And a subset of these styles are being overridden by the `classic.css` styles due to specificity or loading order:

image

Seems to me this is still not working quite as it should, unless the expectation is that I, as a theme author, explicitly dequeue the classic styles.

WP 6.0.2
Gutenberg plugin 14.3.0

**Comment by scruffian** [comment]
@ethanclevenger91 Thanks, that's a great spot. I have a fix over here: https://github.com/WordPress/gutenberg/pull/45063

**Comment by oandregal** [comment]
There's a follow-up in core to remove this stylesheet https://core.trac.wordpress.org/ticket/56990

**Pull request merged**
